### PR TITLE
Fix error when renaming

### PR DESCRIPTION
**Pull request by Colengms**
For enum values, the "!" operator will indicate true for enum values that are 0, in additional to null or undefined.

Switch checks to use `=== undefined` instead.

### CODE REVIEW

File: Extension/src/LanguageServer/referencesTreeDataProvider.ts [hunk 1]
```diff
@@ -121,7 +121,7 @@ export class ReferencesTreeDataProvider implements vscode.TreeDataProvider<TreeN
                 if (!element.referenceText) {
```
Comment (bobbrow) [inline Extension/src/LanguageServer/referencesTreeDataProvider.ts]: empty string will also evaluate to false. Is that something else we should be checking?

File: Extension/src/LanguageServer/referencesTreeDataProvider.ts [hunk 2]
```diff
@@ -121,7 +121,7 @@ export class ReferencesTreeDataProvider implements vscode.TreeDataProvider<TreeN
...
                     throw new Error("Null or undefined referenceText in getTreeItem()");
                 }
-                if (!element.referenceType) {
+                if (element.referenceType === undefined) {
                     throw new Error("Null or undefined referenceType in getTreeItem()");
```
Comment (bobbrow) [inline Extension/src/LanguageServer/referencesTreeDataProvider.ts]: Can you assign null to a `?` property?  I think it can only be `undefined`

Comment (sean-mcmanus) [reply Extension/src/LanguageServer/referencesTreeDataProvider.ts]: Maybe we could enable https://palantir.github.io/tslint/rules/no-null-keyword/ later on.

**Review by sean-mcmanus — APPROVED** [review]

**Review by sean-mcmanus — CHANGES_REQUESTED** [review]
Could you update the changelog with the new date?

File: Extension/src/LanguageServer/referencesTreeDataProvider.ts (hunk 1 above)
Comment (Colengms) [reply Extension/src/LanguageServer/referencesTreeDataProvider.ts]: I don't think we have a lot of cases in which a blank string would be valid.  I addressed some potential cases.

File: Extension/src/LanguageServer/referencesTreeDataProvider.ts (hunk 2 above)
Comment (Colengms) [reply Extension/src/LanguageServer/referencesTreeDataProvider.ts]: I corrected those error messages.  Hypothetically, it's possible for external code (i.e. VS Code APIs) to return null or undefined from APIs that do not indicate they might.  (i.e. for templates  like get<type> in settings, if we support a null setting and don't use get<type | null> when we call it, we may still get null back).

**Comment by Colengms** [comment]
> Could you update the changelog with the new date?

Sure.  I'll update it with the new year, also.  :)

**Review by sean-mcmanus — APPROVED** [review]

**Comment by sean-mcmanus** [comment]
> Could you update the changelog with the new date?
> 
> Sure.  I'll update it with the new year, also.  :)

Good catch!

**Pull request merged**